bridge/setu/listener/heimdall.go

```diff
@@ -0,0 +1,190 @@
+package listener
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/RichardKnop/machinery/v1/tasks"
+	"github.com/maticnetwork/bor/core/types"
+	"github.com/maticnetwork/heimdall/helper"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	checkpointTypes "github.com/maticnetwork/heimdall/checkpoint/types"
+	clerkTypes "github.com/maticnetwork/heimdall/clerk/types"
+)
+
+const (
+	heimdallLastBlockKey = "heimdall-last-block" // storage key
+)
+
+// HeimdallListener - Listens to and process events from heimdall
+type HeimdallListener struct {
+	BaseListener
+}
+
+// NewHeimdallListener - constructor func
+func NewHeimdallListener() *HeimdallListener {
+	return &HeimdallListener{}
+}
+
+// Start starts new block subscription
+func (hl *HeimdallListener) Start() error {
+	hl.Logger.Info("Starting")
+
+	// create cancellable context
+	_, cancelSubscription := context.WithCancel(context.Background())
```

Comment:
Is this context being used anywhere?

Reply:
Yeah...not required for heimdall listener.